### PR TITLE
[processor/spanmetrics] Deprecate spanmetrics processor.

### PR DESCRIPTION
**Pull request by kovrus**
**Description:** <Describe what has changed.>

Deprecate the spanmetrics processor in favor of the spanmetrics connector.

**Link to tracking Issue:** <Issue number if applicable>
https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/19736

**Documentation:** <Describe the documentation added.>
`spanmetrics` processor readme.

### CODE REVIEW

**Comment by kovrus** [comment]
cc: @albertteoh

**Comment by djaglowski** [comment]
@kovrus, the stability level is also defined in code [here](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/cd146d550ac893312aa03fc9c7e6534804606498/processor/spanmetricsprocessor/factory.go#L31). We should update that as well.

File: .chloggen/spanmetrics-deprecate-processor.yaml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: deprecation
+
+# The name of the component, or a single word describing the area of concern, (e.g. filelogreceiver)
+component: spanmetricsprocessor
+
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: Deprecate the `spanmetrics` processor in favour if the `spanmetrics` connector.
```
Comment (albertteoh) [inline .chloggen/spanmetrics-deprecate-processor.yaml:8]:
```suggestion
note: Deprecate the `spanmetrics` processor in favour of the `spanmetrics` connector.
```

**Review by albertteoh — APPROVED** [review]

**Review by mx-psi — CHANGES_REQUESTED** [review]
We should also deprecate the go module, see https://go.dev/ref/mod#go-mod-file-module-deprecation

**Review by mx-psi — APPROVED** [review]

**Review by atoulme — APPROVED** [review]

**Pull request merged**